.buildkite/pipeline.yml

```
@@ -1,3 +1,66 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+
+env:
+  SETUP_GVM_VERSION: 'v0.5.0'
+  DOCKER_COMPOSE_VERSION: '1.25.5'
+  ELASTIC_PACKAGE_COMPOSE_DISABLE_ANSI: "true"
+  KIND_VERSION: 'v0.17.0'
+  K8S_VERSION: 'v1.26.0'
+
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - group: "Check and build"
+    key: "tests"
+    steps:
+      - label: ":go: Run check-ci"
+        key: check-ci
+        command: "make check-ci"
+        agents:
+          image: "golang:1.20.3"
+          cpu: "8"
+          memory: "4G"
+
+      - label: ":go: Run local build"
+        key: make-local
+        command: "make local"
+        agents:
+          image: "golang:1.20.3"
+          cpu: "8"
+          memory: "4G"
+
+  - group: "Run tests"
+    steps:
+      - label: "Run unit test"
+        key: unit-test
+        commands:
+          - "chmod +x .buildkite/scripts/setenv.sh"
+          - ".buildkite/scripts/setenv.sh with-go"
+          - "make test-unit"
+          - "make junit-report"
+          - "buildkite-agent artifact upload build/*.xml"
+        agents:
+          provider: "gcp"
```

the last command set can be specified using `artifact_paths` key:

```suggestion
      - label: "Run unit test"
        key: unit-test
        commands:
          - "chmod +x .buildkite/scripts/setenv.sh"
          - ".buildkite/scripts/setenv.sh with-go"
          - "make test-unit"
          - "make junit-report"
        agents:
          provider: "gcp"
        artifact_paths:
          - build/*.xml
```

---

The command step doesn't support downloading the artifacts, but I need to use it in the next steps:
https://buildkite.com/docs/pipelines/artifacts
That's why I use the Buildkite agent

---

Yes, you are right, mentioned you solution is also working properly, let's use your way because of the same approach.
Thank you!